torch/nn/parallel/distributed.py

```diff
@@ -315,6 +315,28 @@ class DistributedDataParallel(Module):
                          are getting different gradients, which should not
                          happen if DistributedDataParallel is correctly used.
                          (default: ``False``)
+        gradient_as_bucket_view: this is a prototype feature. When set to ``True``,
+                      gradients will be views pointing to different offsets of
+                      allreduce communication buckets. This can reduce peak memory
+                      usage, where the saved memory size will be equal to the total
+                      grads size. Moreover, it avoids the overhead of copying
+                      between gradients and allreduce communication buckets.
+                      When gradients are views, "detach_()" cannot be called on the
+                      gradients. If hitting such errors, please fix it by referring to
+                      the :meth:torch.optim.Optimizer.zero_grad function in
+                      "torch/optim/optimizer.py" as the solution.
+                      Warning! It is also found that 'gradient_as_bucket_view = true'
```

Comment:
warnings are placed in separate blocks. Please search for `.. warning::` in this file.